src/macports1.0/macports_util.tcl

```diff
@@ -35,6 +35,36 @@ package provide macports_util 1.0
 # Provide some global utilities
 
 namespace eval macports_util {
+    ##################
+    # Public methods #
+    ##################
+
+    ##
+    # Given a list of maintainers as recorded in a Portfile, return an
+    # unobscured version that uses proper email addresses.
+    #
+    # @param list A list of obscured maintainers
+    # @return An unobscured list of maintainers
+    proc unobscure_maintainers {list} {
+        set result {}
+        foreach maintainer $list {
+            if {[string first "@" $maintainer] < 0} {
+                # No "@" in the maintainer, it's either an obfuscated email, or
+                # a keyword
+
+                set colonpos [string first ":" $maintainer]
+                if {$colonpos >= 0} {
+                    # Cut and re-assemble
+                    set maintainer "[string range $maintainer $colonpos+1 end]@[string range $maintainer 0 $colonpos-1]"
```

Comment:
`[split ...]` does not have a maximum number of splits, though, so if you want to get this right it would be

```tcl
set user [join [lassign [split $maintainer :] domain] :]
```

which isn't any shorter either. Nevertheless, that's going to be in the next version of this PR.